src/Text/Layout/Table.hs

```diff
@@ -232,37 +232,41 @@ tableLines :: Cell a
            -> [RowGroup a] -- ^ Rows which form a cell together
            -> [String]
 tableLines specs TableStyle { .. } header rowGroups =
-    topLine : addHeaderLines (rowGroupLines ++ [bottomLine])
+    maybe id (:) optTopLine . addHeaderLines $ maybe id (\b -> (++[b])) optBottomLine rowGroupLines
   where
     -- Helpers for horizontal lines that will put layout characters arround and
     -- in between a row of the pre-formatted grid.
 
-    -- | Generate columns filled with 'sym'.
-    fakeColumns sym
-                  = map (`replicateCharB` sym) colWidths
+    -- | Generate columns filled with 'sym', or blank spaces if 'sym' is of width 0.
+    fakeColumns sym = map replicateSym colWidths
+      where
+        replicateSym w = concat $ replicate q sym' ++ [take r sym']
+          where
+            (q, r) = w `quotRem` l
+        (sym', l) = let l' = length sym in if l' == 0 then (" ", 1) else (sym, l')
 
     -- Horizontal seperator lines that occur in a table.
-    topLine       = hLineDetail realTopH realTopL realTopC realTopR $ fakeColumns realTopH
-    bottomLine    = hLineDetail groupBottomH groupBottomL groupBottomC groupBottomR $ fakeColumns groupBottomH
-    groupSepLine  = hLineDetail groupSepH groupSepLC groupSepC groupSepRC $ fakeColumns groupSepH
-    headerSepLine = hLineDetail headerSepH headerSepLC headerSepC headerSepRC $ fakeColumns headerSepH
+    optTopLine       = optHorizontalDetailLine realTopH realTopL realTopC realTopR $ fakeColumns realTopH
+    optBottomLine    = optHorizontalDetailLine groupBottomH groupBottomL groupBottomC groupBottomR $ fakeColumns groupBottomH
+    optGroupSepLine  = optHorizontalDetailLine groupSepH groupSepLC groupSepC groupSepRC $ fakeColumns groupSepH
+    optHeaderSepLine = optHorizontalDetailLine headerSepH headerSepLC headerSepC headerSepRC $ fakeColumns headerSepH
 
     -- Vertical content lines
-    rowGroupLines =
-        intercalate [groupSepLine] $ map (map (hLineContent groupV) . applyRowMods . rows) rowGroups
+    rowGroupLines = maybe concat (\seps -> intercalate [seps]) optGroupSepLine
+           $ map (map (horizontalContentLine groupL groupC groupR) . applyRowMods . rows) rowGroups
```

Comment:
Sorry to nit-pick, but please indent either by 4 spaces or the arguments.  For example:

    rowGroupLines =
        maybe concat (\seps -> intercalate [seps]) optGroupSepLine
            $ map (map (horizontalContentLine groupL groupC groupR) . applyRowMods . rows)
                  rowGroups

Or introduces names:

    rowGroupLines = maybe concat (\seps -> intercalate [seps]) optGroupSepLine linesPerRowGroup
    linesPerRowGroup = map rowGroupToLines rowGroups
    rowGroupToLines = map (horizontalContentLine groupL groupC groupR) . applyRowMods . rows

Actually, I prefer the latter more because it splits the expression into understandable chunks.